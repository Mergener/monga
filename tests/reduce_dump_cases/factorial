function factorial(x : int): int {
	var n : int;

	n = 1;
	while (x > 1) {
		n = n * x;
		x = x - 1;
	}

	return n;
}

function main() {
	var i : int;

	i = getInput();

	print(factorial(i));
}