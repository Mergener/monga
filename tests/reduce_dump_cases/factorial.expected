type r1
parameter r1
parameters r1
opt_parameters r2
type r1
opt_ret_type r2
type r1
def_variable r1
block r8
statements r1
var r1
numeral r2
exp_primary r2
exp_postfix r1
exp_unary r1
exp_multiplicative r1
exp_additive r1
exp_conditional r1
exp r1
block r3
statements r2
var r1
exp_primary r3
exp_postfix r1
exp_unary r1
exp_multiplicative r1
exp_additive r1
numeral r2
exp_primary r2
exp_postfix r1
exp_unary r1
exp_multiplicative r1
exp_additive r1
cond_primary r6
cond_not r1
cond_and r1
cond_or r1
cond_or r1
cond_primary r1
cond_not r1
cond_and r1
cond_or r1
cond_or r1
var r1
var r1
exp_primary r3
exp_postfix r1
exp_unary r1
exp_multiplicative r1
var r1
exp_primary r3
exp_postfix r1
exp_unary r1
exp_multiplicative r2
exp_additive r1
exp_conditional r1
exp r1
block r3
statements r1
var r1
var r1
exp_primary r3
exp_postfix r1
exp_unary r1
exp_multiplicative r1
exp_additive r1
numeral r2
exp_primary r2
exp_postfix r1
exp_unary r1
exp_multiplicative r1
exp_additive r3
exp_conditional r1
exp r1
block r3
statements r2
opt_statements r2
block r1
block r2
statements r2
var r1
exp_primary r3
exp_postfix r1
exp_unary r1
exp_multiplicative r1
exp_additive r1
exp_conditional r1
exp r1
opt_exp r2
block r4
statements r2
opt_statements r2
block r1
def_function r1
definition r2
definitions r1
opt_parameters r1
opt_ret_type r1
type r1
def_variable r1
block r8
statements r1
var r1
opt_exps r1
call r1
exp_primary r4
exp_postfix r1
exp_unary r1
exp_multiplicative r1
exp_additive r1
exp_conditional r1
exp r1
block r3
statements r2
var r1
exp_primary r3
exp_postfix r1
exp_unary r1
exp_multiplicative r1
exp_additive r1
exp_conditional r1
exp r1
exps r1
opt_exps r2
call r1
exp_primary r4
exp_postfix r1
exp_unary r1
exp_multiplicative r1
exp_additive r1
exp_conditional r1
exp r1
exps r1
opt_exps r2
call r1
block r5
statements r2
opt_statements r2
block r1
def_function r1
definition r2
definitions r2
module r1
